### DISCUSSION TITLE
RTL Support?

### DISCUSSION THREAD

**Discussion by AmirAliRashidi** [illright/attractions#359] [Q&A]
Hi, I read the document and thats awesome, but does it support RTL?

**Comment by aabounegm** [comment]
Hi. Can you please expand on what did you try and what shortcomings did you find so far? Attractions is just a components library without a layout system, so which components did you try and find to not work so well with RTL?

I tried opening one of our documentation pages and add the `dir="rtl"` attribute to the `html` tag just to see how it looks like, and the default browser behavior seemed to be working well enough. I did find a few shortcomings when it comes to properties that use `right` or `left` in their names (such as `padding-right` and `border-left`), and I see that most of them can be fixed by replacing them with the equivalent `start` and `end` properties (such as `padding-inline-end`) but we need to check how that would hurt browser compatibility.
In the meantime, do you have examples of instances where the library wasn't particularly RTL-friendly?

**Reply by kefahi** [reply]
الحقيقة أن هذه المكتبة أعجبتني كثيرا. فيها بساطة ووضوح في التصميم. وأفكر في استخدامها لتطوير موقع عربي يناسب الهواتف.
لذلك استوقفني سؤال الأخ.
هل تنصحني بتجربة المكتبة في تطبيق عربي، وهل تظن أنه سيكون من السهل حل المشاكل؟
وما هي آخر أخبار هذه المكتبة؟ هل هناك إصدارات جديدة لها تفكرون في إطلاقها؟
ولكم جزيل الشكر على المجهود الرائع،

**Reply by aabounegm** [reply]
<p dir="rtl">
@kefahi شكرا لتعليقك ودعمك. في الوقت الحالي لا أظن أنها مناسبة بما فيه الكفاية، ولكني أنوي العمل على هذا. بإمكانك أن تجربها وتخبرنا بالمشاكل التي تواجهها، وأعتقد أنها بإذن الله بسيطة وسهلة الحل.
حاليا توقفنا عن التطوير بعض الشيء لأسباب شخصية (مثل الامتحانات في الجامعة)، لكننا ما زلنا نتابع أي إسهامات ونرحب بها. قريبا إن شاء الله أعود لها وأحل هذه المشكلة بعد التحويل الكبير لـSvelteKit.

تعليق صغير: المطور الآخر لا يفهم العربية، فرجاءً استعمل الانجليزية إن أمكن.
</p>

Short summary: Many components are not currently suitable for RTL, but the changes should be relatively straightforward. We are somewhat busy at the time being, but contributions are welcome, and I plan to tackle this issue once I find time and complete the SvelteKit migration.
